src/components/SettingsModal.vue

```diff
@@ -1,36 +1,52 @@
 <script setup lang="ts">
+import useSettings from '@/composables/useSettings';
+import ToggleSwitch from './ToggleSwitch.vue';
 
+const emit = defineEmits(['toggleDevMode', 'toggleHardMode'])
+
+const { devMode, hardMode } = useSettings()
+
+function toggleDevMode() {
+  emit('toggleDevMode')
+}
+
+function toggleHardMode() {
+  emit('toggleHardMode')
+}
```

Comment:
You can use `toggleDevMode` and `toggleHardMode` directly from `useSettings` here and remove the emitted events